contrib/drivers/filesharedrivers/netapp/ontap_nas.go

```diff
@@ -0,0 +1,295 @@
+	// Read NetApp ONTAP config file
+	d.conf = &ONTAPConfig{}
+
+	p := config.CONF.OsdsDock.Backends.NetappOntapNas.ConfigPath
+	if "" == p {
+		p = defaultConfPath
+	}
+	if _, err := Parse(d.conf, p); err != nil {
+		return err
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			log.Error("unable to instantiate ontap backend.")
+		}
+	}()
+
+	empty := ""
+	config := &drivers.OntapStorageDriverConfig{
+		CommonStorageDriverConfig: &drivers.CommonStorageDriverConfig{
+			Version:           d.conf.Version,
+			StorageDriverName: StorageDriverName,
+			StoragePrefixRaw:  json.RawMessage("{}"),
+			StoragePrefix:     &empty,
+		},
+		ManagementLIF: d.conf.ManagementLIF, // This is Storage VM (SVM) IP
+		DataLIF:       d.conf.DataLIF,       // This is the data LIF for NFS/CIFS access
+		SVM:           d.conf.Svm,
+		Username:      d.conf.Username,
+		Password:      d.conf.Password,
+	}
+	marshaledJSON, err := json.Marshal(config)
+	if err != nil {
+		log.Fatal("unable to marshal ONTAP config:  ", err)
+	}
+	configJSON := string(marshaledJSON)
+
+	// Convert config (JSON or YAML) to JSON
+	configJSONBytes, err := yaml.YAMLToJSON([]byte(configJSON))
+	if err != nil {
+		err = fmt.Errorf("invalid config format: %v", err)
+		return err
+	}
+	configJSON = string(configJSONBytes)
+
+	// Parse the common config struct from JSON
+	commonConfig, err := drivers.ValidateCommonSettings(configJSON)
+	if err != nil {
+		err = fmt.Errorf("input failed validation: %v", err)
+		return err
+	}
+
+	// Create the NAS Storage driver instance from the trident driver
+	d.nasStorageDriver = &ontap.NASStorageDriver{
+		Config: *config,
+	}
+
+	// Initialize the driver.
+	if err = d.nasStorageDriver.Initialize(driverContext, configJSON, commonConfig); err != nil {
+		log.Errorf("could not initialize storage driver (%s). failed: %v", commonConfig.StorageDriverName, err)
+		return err
+	}
+	log.Infof("storage driver (%s) initialized successfully.", commonConfig.StorageDriverName)
+
+	return nil
+}
+
+func (d *NASDriver) Unset() error {
+	//driver to clean up and stop any ongoing operations.
+	d.nasStorageDriver.Terminate()
+	return nil
+}
+
+// Method to create a fileshare on the filer
+func (d *NASDriver) CreateFileShare(opt *pb.CreateFileShareOpts) (vol *model.FileShareSpec, err error) {
+	var name = opt.GetName()
+	volConfig := d.GetVolumeConfig(name, opt.GetSize())
+	storagePool := &storage.Pool{
+		Name:               opt.GetPoolName(),
+		StorageClasses:     make([]string, 0),
+		Attributes:         make(map[string]sa.Offer),
+		InternalAttributes: make(map[string]string),
+	}
+
+	// Here the storage pool is the aggregate on the Netapp Filer
+	// This Create call will create a volume and mount to be used as
+	// the export/share
+	err = d.nasStorageDriver.Create(volConfig, storagePool, make(map[string]sa.Request))
+	if err != nil {
+		log.Errorf("create nas fileshare (%s) failed: %v", opt.GetId(), err)
+		return nil, err
+	}
+	var exportLocation []string
```

Comment:
Two question:
1. why we need exportLocation as array or map (Overall it's just a string)
2. Don't think, server  ip address need to verify the IP format etc. ?

Reply:
1. The current Fileshare model supports only array, not string
2. This IP is the DataLIF provided (The dataLIF IP is the NFS server IP of the SVM), so this is already validated, because if this is wrong, Setup() would have already failed.